File: services/ordering/Ordering.Domain/Orders/Order.cs
namespace Ordering.Domain.Orders;

public sealed class Order
{
    public Guid Id { get; set; }

    public string CustomerName { get; set; } = string.Empty;

    public string CustomerEmail { get; set; } = string.Empty;

    public OrderStatus Status { get; set; }

    public long TotalAmountMinor { get; set; }

    public string Currency { get; set; } = string.Empty;

    public DateTimeOffset CreatedAt { get; set; }

    public DateTimeOffset UpdatedAt { get; set; }

    public List<OrderItem> Items { get; set; } = [];
}

File: frontend/angular-app/src/app/features/orders/models/order.ts
export type OrderItem = {
  id: string;
  productId: string;
  productVariantId: string;
  sku: string;
  productName: string;
  variantName: string;
  unitPriceAmountMinor: number;
  currency: string;
  quantity: number;
  lineTotalAmountMinor: number;
  inventoryReservationId: string | null;
};

export type OrderDetails = {
  id: string;
  customerName: string;
  customerEmail: string;
  status: string;
  totalAmountMinor: number;
  currency: string;
  createdAt: string;
  updatedAt: string;
  items: OrderItem[];
};

export type CreateOrderItemRequest = {
  productVariantId: string;
  quantity: number;
};

export type CreateOrderRequest = {
  customerName: string;
  customerEmail: string;
  items: CreateOrderItemRequest[];
};
